server/src/main/java/org/apache/druid/client/indexing/ClientCompactionRunnerInfo.java

```diff
@@ -163,17 +172,26 @@ public static CompactionConfigValidationResult validatePartitionsSpecForMSQ(
       Map<String, DimensionSchema> dimensionSchemaMap = dimensionSchemas.stream().collect(
           Collectors.toMap(DimensionSchema::getName, Function.identity())
       );
-      Optional<String> nonStringDimension = ((DimensionRangePartitionsSpec) partitionsSpec)
+      Optional<String> unsupportedDimension = ((DimensionRangePartitionsSpec) partitionsSpec)
           .getPartitionDimensions()
           .stream()
-          .filter(dim -> !ColumnType.STRING.equals(dimensionSchemaMap.get(dim).getColumnType()))
+          .filter(dim -> !ColumnType.STRING.equals(dimensionSchemaMap.get(dim).getColumnType())
+                         || multiValuedDimensions != null && multiValuedDimensions.contains(dim))
```

Comment:
For clearer code, let's leave the whole check for non string dimension as-is.
Add a separate loop for checking multi-valued dimensions.

Reply:
Done.